tests/unit/test_dimension.py

```diff
@@ -1217,6 +1222,26 @@ def but_it_raises_when_insertion_id_not_present(self, _order_dict_prop_):
             _OrderSpec(None, None).insertion_id
         assert str(e.value) == "'insertion_id'"
 
+    def it_knows_the_marginal_specified_as_the_sort_basis(self, _order_dict_prop_):
+        _order_dict_prop_.return_value = {"marginal": "scale_mean"}
+        assert _OrderSpec(None, None).marginal == MARGINAL.SCALE_MEAN
+
+    def but_it_raises_when_marginal_field_is_not_present(self, _order_dict_prop_):
+        _order_dict_prop_.return_value = {}
+        with pytest.raises(KeyError) as e:
+            _OrderSpec(None, None).marginal
+        assert str(e.value) == "'marginal'"
```

Comment:
If we're going to bother to test an exception the exception message should be more descriptive. I'm thinking this is a case that's unlikely to arise once we're done with migration because all valid values will be implemented. I would just leave this test out.